### ISSUE TITLE
Request to add OpenAI third-party base_url and manual input for custom model settings.

### ISSUE THREAD

**Issue by popcion** [ogkalu2/comic-translate#161]
![image](https://github.com/user-attachments/assets/ddb5c16a-9228-4b1f-a767-14bbd2589a4f)
https://sider.ai/
https://github.com/songquanpeng/one-api
We can use a third-party base_url to relay all non-OpenAI models in the OpenAI format.
Here is a api guidance.
https://chatanywhere.apifox.cn/api-92222076

![image](https://github.com/user-attachments/assets/b7cf02c8-9cda-44ac-b4f2-a91a0a24fee1)

**Comment by iSams123** [comment]
Yeah, I would love to use compatible OpenAI API providers (with custom **base_url**) besides the official one

**Comment by ogkalu2** [comment]
@iSams123 @popcion Now possible
